sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventData.java

```diff
@@ -73,16 +77,24 @@
      * Creates an event containing the {@code body}.
      *
      * @param body The data to set for this event.
+     *
      * @throws NullPointerException if {@code body} is {@code null}.
      */
     public EventData(byte[] body) {
-        this(BinaryData.fromBytes(Objects.requireNonNull(body, "'body' cannot be null.")));
+        this.systemProperties = new SystemProperties();
```

Comment:
Since AmqpAnnotatedMessage exposes byte[] as the body, I felt like it was best to store it as such rather than BinaryData that we were using before to stop a lot of the conversions to and from BinaryData.

Reply:
@conniey, @yijun:  This looks like `SystemProperties` is a copy of data that also appears on the `AmqpAnnotatedMessage` instance.   Since the `AmqpAnnotatedMessage` instance is fully mutable, if callers change the data there, then the `EventData` projection is no longer accurate and you've got drift....

Am I overlooking something?